src/modules/PaintBucket/PaintBucket.js

```diff
@@ -0,0 +1,64 @@
+module.exports = exports = function(pixels, startingX, startingY, fillColor){
+
+      fillColor = fillColor.split(" ");
+      var x = startingX,
+          y = startingY,
+          height = pixels.shape[1],
+          width = pixels.shape[0],
+          r = pixels.get(x, y, 0),
+          g = pixels.get(x,y,1),
+          b = pixels.get(x,y,2),
+          a = pixels.get(x, y, 3),
+          queuex = [x],
+          queuey = [y],
+          curry, currx,
+          north,
+          south,
+          n,
+          tolerance = 5;
+
+      function isSimilar(currx, curry){
+        if (pixels.get(currx, curry, 0) > r*(1 - tolerance/100) && pixels.get(currx, curry, 0) < r*(1 + tolerance/100) &&
+            pixels.get(currx, curry, 1) > g*(1 - tolerance/100) && pixels.get(currx, curry, 1) < g*(1 + tolerance/100) &&
+            pixels.get(currx, curry, 2) > b*(1 - tolerance/100) && pixels.get(currx, curry, 2) < b*(1 + tolerance/100) &&
```

Comment:
Also maybe you can assign 1-tolerance/100 and 1+tolerance/100 to different vars?

Reply:
I have made the changes. Thanks!